### PR TITLE
[IA-4659] Rename some IA models

### PR DESCRIPTION
**Pull request by jdcanas**
This is a pretty simple PR that just renames a few things and converts one small file to typescript

Specifically:
- Renames `ListAppResponse/GetAppResponse` to `ListAppItem/GetAppItem` for consistency with other Leo models
- Renames RuntimeManager to something more sensible
- converts RuntimeManager to typescript and fixes existing bugs that typescript identified

### CODE REVIEW

File: src/analysis/AnalysisNotificationManager.ts [hunk 1]
```diff
@@ -120,30 +141,43 @@ const AppErrorNotification = ({ app }) => {
...
-    const runtime = getCurrentRuntime(runtimes) || {};
-    const prevRuntime = _.last(_.sortBy('auditInfo.createdDate', _.remove({ status: 'Deleting' }, prevRuntimes))) || {};
+    const runtime: Runtime | undefined = getCurrentRuntime(runtimes);
+    const prevRuntime: Runtime | undefined = _.last(
+      _.sortBy('auditInfo.createdDate', _.remove({ status: 'Deleting' }, prevRuntimes))
+    );
     const twoMonthsAgo = _.tap((d) => d.setMonth(d.getMonth() - 2), new Date());
     const welderCutOff = new Date('2019-08-01');
-    const createdDate = new Date(runtime?.createdDate);
+    const createdDate = runtime ? new Date(runtime.auditInfo.createdDate) : new Date();
```
Comment (jdcanas) [inline src/analysis/AnalysisNotificationManager.ts:163]: as always, typescript finds bugs. This code was not working before.... (notice the incorrect `runtime.createdDate` changed to `runtime.auditInfo.createdDate`)

File: src/analysis/AnalysisNotificationManager.ts [hunk 2]
```diff
@@ -120,30 +141,43 @@ const AppErrorNotification = ({ app }) => {
...
+  apps: App[];
+}
+
+export const AnalysisNotificationManager = (props: AnalysisNotificationManagerProps) => {
+  const { namespace, name, runtimes, apps } = props;
+  const prevRuntimes: Runtime[] = usePrevious(runtimes) || [];
+  const prevApps: App[] = usePrevious(apps) || [];
 
   useEffect(() => {
-    const runtime = getCurrentRuntime(runtimes) || {};
```
Comment (jdcanas) [inline src/analysis/AnalysisNotificationManager.ts:128]: This attempt to make runtime not undefined wasn't entirely correct... we defaulted it to an empty object which is not the same as a runtime object. There were a lot of bugs surrounding this that typing it as the return of `getCurrentRuntime` (`Runtime | undefined`) trivially fixed.

File: src/analysis/AnalysisNotificationManager.ts [hunk 3]
```diff
@@ -120,30 +141,43 @@ const AppErrorNotification = ({ app }) => {
   ]);
 };
 
-function RuntimeManager({ namespace, name, runtimes, apps }) {
```
Comment (jdcanas) [inline src/analysis/AnalysisNotificationManager.ts:123]: This was still using the old function syntax we ditched a while ago as well

**Review by nawatts — APPROVED** [review]
Approving changes to AJ owned files.

**Review by aherbst-broad — APPROVED** [review]

**Review by LizBaldo — APPROVED** [review]
LGTM, thanks 👍

**Review by mlilynolting — APPROVED** [review]
looks good. the one change i'm unsure about is GetAppItem; the context of getApp is already one app, not many, so GetAppResponse would still be okay imo

**Pull request merged**